### ISSUE TITLE
[BUG] Cloudflare Worker unsupported based on documentation

### ISSUE THREAD

**Issue by lorenzodejong** [inngest/inngest-js#617]
**Describe the bug**
When initialising a new Cloudflare Worker project using the Wrangler CLI, the default documented setup from the [documentation](https://www.inngest.com/docs/sdk/serve#framework-cloudflare) does not work as intended.

**To Reproduce**

Cloudflare Worker based on https://developers.cloudflare.com/workers/get-started/guide/

1. Create a new Worker by running `npm create cloudflare@latest` (choose Hello world starter and enable TypeScript)
2. `npm i inngest`

Setup a client:
```typescript
// src/inngest/client.ts
import { Inngest } from 'inngest';

export const inngest = new Inngest({ id: 'example' });
```

Setup a function:
```typescript
// src/inngest/functions.ts
import { inngest } from './client';

export const helloWorld = inngest.createFunction({ id: 'hello-world' }, { event: 'test/hello.world' }, async ({ event, step }) => {
	await step.sleep('wait-a-moment', '1s');
	return { event, body: 'Hello, World!' };
});
```

Configure the worker based on the [documentation](https://www.inngest.com/docs/sdk/serve#framework-cloudflare)
```typescript
// src/index.ts
import { serve } from "inngest/cloudflare";
import { inngest } from "./inngest/client";
import { helloWorld } from "./inngest/functions";

export const onRequest = serve({
  client: inngest,
  functions: [helloWorld],
});
```

Run the worker using `npm dev`.

**Observed behavior:**
The Worker doesn't handle the setup, it only seems to be compatible with Cloudflare Pages.

It does work when configuring the Worker using the method below, is this the recommended way of configuration?

```typescript
import { serve } from 'inngest/cloudflare';
import { inngest } from './inngest/client';
import { helloWorld } from './inngest/functions';

export default {
	async fetch(request, env): Promise<Response> {
		return serve({
			client: inngest,
			functions: [helloWorld],
		})({ env: env as Record<string, string>, request });
	},
} satisfies ExportedHandler<Env>;
```

**Expected behavior**
The Cloudflare Worker is correctly setup using the documented `inngest/cloudflare` setup OR the documentation is adjusted to reflect the intended configuration.

**Code snippets / Logs / Screenshots**
n/a

**System info (please complete the following information):**

- OS: Mac
- pnpm 8.15.3
- Framework Cloudflare Workers (wrangler `^3.0.0`)
- Platform Cloudflare

**Additional context**
n/a

**Comment by jpwilliams** [comment]
Thanks for the report, @lorenzodejong!

We'll fix both:

- Adjust `"inngest/cloudflare"` to also handle the Workers signature, so we can do `export default serve()` (or `export default { ...serve(), ... }`)
- Adjust docs to highlight the different exports needed

**Comment by lorenzodejong** [comment]
Amazing, thanks @jpwilliams!

Another thing i noticed **(if you would like me to create a separate issue let me know, i'd be happy to do so)**:

With the current implementation the `--remote` development environment using the Wrangler CLI seems to be unsupported with the local dev server of Inngest.

Basically what you can do with Wrangler is run the following: `wrangler dev --remote`, which will deploy your Worker to a temporary instance in Cloudflare. This will allow you to connect to any bindings which are [unsupported in local development](https://developers.cloudflare.com/workers/testing/local-development/#supported-resource-bindings-in-different-environments). This opens up a proxy on your localhost, which would theoretically allow you to invoke Inngest events locally forwarding it to the instance running in Cloudflare.

However when trying to register the endpoint to the local dev server of Inngest, it's indicating that the registered environment is detected to be `cloud` instead of `dev` (causing the registration to fail). This makes sense from a technical standpoint, as it's actually running in a cloud environment.

I tried to circumvent this by explicitly setting `isDev` to `true` on the Inngest client. This does allow registration of the instance in the Inngest dev server, however any invoked event will cause an error from Cloudflare: `Error 1003 : Direct IP access not allowed`. It seems like the Inngest dev server is explicitly trying to access a port on the host, which is not allowed from Cloudflare.

**Comment by BernardoSM** [comment]
@lorenzodejong I had the same problem with --remote option from cloudflare.

I have some mTLS bindings that are only available with `--remote`, so I can't use them with inngest right now as I said here: https://github.com/inngest/inngest/issues/1463

**Comment by lorenzodejong** [comment]
@BernardoSM thanks for noticing, since this issue will be resolved from #619 i'm closing this. We can continue the thread from your provided issue.
